Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

```diff
@@ -0,0 +1,294 @@
+
+Next, your library should "*restore*" the contextual baggage, this is performed by setting the baggage task-local value around calling into user code, like this:
+
+```swift
+func handler(request: HTTPRequest) async { 
+  // ... 
+  InstrumentationSystem.instrument.extract(..., into: &baggage, ...)
+  // ... 
+
+  await Baggage.withValue(baggage) {
+    await userCode(request)
+  }
+}
+```
+
+This sets the task-local value `Baggage.current` which is used by [swift-log](https://github.com/apple/swift-log), as well as ``TracerProtocol`` APIs in order to later "*pick up*" the baggage and and e.g. include it in log statements, or start new trace spans using the information stored in the baggage.
+
+> Note: The end goal here being that when end-users of your library write `log.info("Hello")` the logger is able to pick up the contextual baggage information and include the e.g. the `trace-id` in such log statement automatically! This way, every log made during the handling of this request would include the trade-id automatically, e.g. like this: 
+>
+> `12:43:32 info [trace-id=463a...13ad] Logging during handling of a request!`
+
+If your library makes multiple calls to user-code as part of handling the same request, you may consider if restoring the baggage around all of these callbacks is beneficial. For example, if a library had callbacks such as:
+
+```swift
+/// Example library protocol, offering multiple callbacks, all semantically part of the same request handling.
+protocol SampleHTTPHandler {
+  // called immediately when an incoming HTTPRequests headers are available, 
+  // even before the request body has been processed.
+  func requestHeaders(headers: HTTPHeaders) async
+  
+  // Called multiple (or zero) times, for each "part" of the incoming HTTPRequest body.
+  func requestBodyPart(ByteBuffer) async
+}
+```
+
+You may want to restore the baggage once around both those calls, or if that is not possible, restore it every time when calling back into user code, e.g. like this:
+
+```swift
+actor MySampleServer { 
+  
+  var baggage: Baggage = .topLevel
+  var userCode: SampleHTTPHandler
+
+  func onHeaders(headers: HTTPHeaders) async {
+    await Baggage.withValue(self.baggage) { 
+      await userHandler.requestHeaders(headers)
+    }
+  }
+  
+  func onBodyPart(part: ByteBuffer) async { 
+    await Baggage.withValue(self.baggage) { 
+      await userHandler.requestHeaders(headers)
+    }
+  }
+}
+```
+
+While this code is very simple for illustration purposes, and it may seem suprising why there are those two separate places where we need to call into user-code separately. But in practice such situations can happen when using asynchronous network or database libraries which offer their API in terms of such callbacks. Always consider if and when to restore baggage such that it makes sense for the end user.
```

Comment:
```suggestion
While this code is very simple for illustration purposes, and it may seem surprising why there are two separate places where we need to call into user-code separately, in practice such situations can happen when using asynchronous network or database libraries which offer their API in terms of callbacks. Always consider if and when to restore baggage such that it makes sense for the end user.
```